lib/src/lifecycle_module.dart

```diff
@@ -77,12 +77,12 @@ abstract class LifecycleModule {
   Future loadChildModule(LifecycleModule newModule) async {
     newModule.didLoad.listen((_) {
       _childModules.add(newModule);
-      _didLoadChildModuleController.add(newModule);
     });
     newModule.willUnload.listen((_) {
       _childModules.remove(newModule);
     });
     await newModule.load();
+    await _didLoadChildModuleController.add(newModule);
```

Comment:
for readability sake, just completely kill the entire `newModule.didLoad.listen` block so the bottom code chunk can be
```
await newModule.load();
_childModules.add(newModule);
await _didLoadChildModuleController.add(newModule);
```